docs/tutorials/projects/bedtime-songs-bot.md

```diff
@@ -0,0 +1,568 @@
+   }
+   defer f.Close()
+
+
+   streamer, format, err := mp3.Decode(f)
+   if err != nil {
+       logger.Fatal(err)
+   }
+   defer streamer.Close()
+
+
+   speaker.Init(format.SampleRate, format.SampleRate.N(time.Second/10))
+}
+
+// Play a song
+func play(label string, logger golog.Logger) {
+   f, err := os.Open(label + ".mp3")
+   if err != nil {
+       logger.Fatal(err)
+   }
+   defer f.Close()
+
+
+   streamer, _, err := mp3.Decode(f)
+   if err != nil {
+       logger.Fatal(err)
+   }
+   defer streamer.Close()
+
+
+   done := make(chan bool)
+   speaker.Play(beep.Seq(streamer, beep.Callback(func() {
+       done <- true
+   })))
+
+
+   <-done
+}
+
+// Code Sample Connect() Code 
+func main() {
+ logger := golog.NewDevelopmentLogger("client")
+ robot, err := client.New(
+     context.Background(),
+     ".viam.cloud", // Insert your remote address here. Go to the Code Sample tab in the Viam app to find.
+     logger,
+     client.WithDialOptions(rpc.WithCredentials(rpc.Credentials{
+         Type:    utils.CredentialsTypeRobotLocationSecret,
+         Payload: "", // Insert your robot location secret here. Go to the Code Sample tab in the Viam app to find.
+     })),
+ )
+ if err != nil {
+     logger.Fatal(err)
+ }
+
+
+ defer robot.Close(context.Background())
+  
+// Get the shape classifier from the robot 
+ visService, err := vision.FromRobot(robot, "shape-classifier")
+ if err != nil {
+   logger.Error(err)
+ }
+
+ // Initialize the speaker
+ initSpeaker(logger)
+
+
+ // Classifications logic
+ for {
+   for i := 0; i < 3; i++ {
+       visService.ClassificationsFromCamera(context.Background(), "cam",  1, nil)
+   }
+
+
+   classifications, err := visService.ClassificationsFromCamera(context.Background(), "cam", 1, nil)
+   if err != nil {
+       logger.Fatalf("Could not get classifications: %v", err)
+   }
+   if len(classifications) > 0 && classifications[0].Score() > 0.7 {
+       logger.Info(classifications[0])
+       play(classifications[0].Label(), logger)
+   }
+
+
+ }
+}
+```
+
+Save your <file>play-songs.go</file> program with this logic added in.
+Run the code on your personal computer as follows:
+
+```sh {class="command-line" data-prompt="$"}
+go run ~/<my-bedtime-songs-bot-directory>/play-songs.go
+```
+
+The full example source code for <file>play-songs.go</file> is available on [GitHub](https://github.com/viam-labs/bedtime-songs-bot/blob/main/play_songs.go).
+
+Now, as shown below, your smart bedtime songs bot knows to play a song whenever it sees a shape on the camera:
+
+{{<video webm_src="/tutorials/img/bedtime-songs-bot/robot_babysitter.webm" mp4_src="/tutorials/img/bedtime-songs-bot/robot_babysitter.mp4" max-width="500px" alt="A demonstration of the bedtime songs bot is taking place in an office. Tess holds up brightly colored puzzle pieces in front of the camera of a Macbook laptop. As the webcam on the laptop recognizes the puzzle pieces, different songs start to play on the speakers of the computer.">}}
```

Comment:
Can this be a video file that has sound? Since this is such an audio-focused tutorial?

Reply:
I'll see if I can find it tomorrow and remove the editing I did to make it look better from my iphone quality that took away the sound

For now I have absolutely no idea why it keeps on failing these checks (I thought I have recs for poster in front matter?), @npentrel If you have any advice that would be appreciated dearly

Reply:
```suggestion
{{<gif webm_src="/tutorials/img/bedtime-songs-bot/robot_babysitter.webm" mp4_src="/tutorials/img/bedtime-songs-bot/robot_babysitter.mp4" max-width="500px" alt="A demonstration of the bedtime songs bot is taking place in an office. Tess holds up brightly colored puzzle pieces in front of the camera of a Macbook laptop. As the webcam on the laptop recognizes the puzzle pieces, different songs start to play on the speakers of the computer.">}}
```
Ahh here's the problem :D If you change this from "video" to "gif" it all builds properly. If you do change it to a video with sound though, it won't be an autoplaying gif anymore, and then it'll need to say "video" and have a "poster" added into this shortcode too.

Reply:
You don't need "poster" or "image" in the frontmatter though if you keep the webm and mp4s in there. You'd only need the poster inside this shortcode, for the click-to-play video, if you go that route.

Reply:
https://github.com/viamrobotics/docs/blob/main/docs/note.md#regular-videos

Videos need an image to display before they load. You add that on the shortcode. The video on this page and how you add it doesn't interact with the frontmatter specification. 

```suggestion
{{<video webm_src="/tutorials/img/bedtime-songs-bot/robot_babysitter.webm" mp4_src="/tutorials/img/bedtime-songs-bot/robot_babysitter.mp4" max-width="500px" alt="A demonstration of the bedtime songs bot is taking place in an office. Tess holds up brightly colored puzzle pieces in front of the camera of a Macbook laptop. As the webcam on the laptop recognizes the puzzle pieces, different songs start to play on the speakers of the computer." poster="/general/heart.jpg">}}
```
^ this will make it work but you need to get an image from the video. You can use:

```sh
ffmpeg -ss 00:00:05 -i PATH_TO_GIF_OR_VID.mp4 -frames:v 1 PATH_TO_GIF_OR_VID.jpg
```

Reply:
Indeed it doesn't interact; I just mean that 
1. It's an either/or: either this becomes a GIF (which works with the current video that doesn't have sound anyway) OR it becomes a video in which case it needs a poster.
2. Included the sentence about not needing a poster in the frontmatter because we've been messing with that too, and if it stays as a GIF in the frontmatter you don't need a poster do you?